sw/device/tests/keymgr_dpe_key_derivation_test.c

```diff
@@ -42,64 +42,375 @@ static void init_kmac_for_keymgr(void) {
+  generate(&keymgr_dpe, &gen_params);
+  LOG_INFO("KeymgrDpe generated OTBN output from boot stage 0");
 
-  CHECK_STATUS_OK(keymgr_dpe_testutils_wait_for_operation_done(&keymgr_dpe));
+  // Generate SW output from boot stage 0 key.
+  gen_params.slot_src_sel = 1;
+  gen_params.sideload_key = false;  // SW key
+  gen_params.key_dest = kDifKeymgrDpeKeyDestNone;
+  gen_params.version = 0;
+  gen_params.salt[7] = 0x72d5886b;
+  gen_params.salt[6] = 0x4e359e52;
+  gen_params.salt[5] = 0x0d7ff336;
+  gen_params.salt[4] = 0x267773cf;
+  gen_params.salt[3] = 0x00c7d10c;
+  gen_params.salt[2] = 0x6dea4fb9;
+  gen_params.salt[1] = 0x77fa328a;
+  gen_params.salt[0] = 0x15779805;
+  generate(&keymgr_dpe, &gen_params);
+  LOG_INFO("KeymgrDpe generated SW output from boot stage 0");
 
-  dif_keymgr_dpe_generate_params_t genparams;
-  genparams.key_dest = kDifKeymgrDpeKeyDestAes;
-  genparams.sideload_key = false;
+  // Generate KMAC output from the boot stage 0 key.
+  gen_params.slot_src_sel = 1;
+  gen_params.sideload_key = true;  // HW key
+  gen_params.key_dest = kDifKeymgrDpeKeyDestKmac;
+  gen_params.version = 0;
+  gen_params.salt[7] = 0x78ad5715;
+  gen_params.salt[6] = 0x508680d4;
+  gen_params.salt[5] = 0xc7f825b2;
+  gen_params.salt[4] = 0xa7924b8d;
+  gen_params.salt[3] = 0x0906825f;
+  gen_params.salt[2] = 0x77cf81a3;
+  gen_params.salt[1] = 0xd63d89bd;
+  gen_params.salt[0] = 0x88fd3697;
+  generate(&keymgr_dpe, &gen_params);
+  LOG_INFO("KeymgrDpe generated KMAC output from boot stage 0");
 
-  genparams.salt[0] = 0xd;
-  genparams.salt[1] = 0xe;
-  genparams.salt[2] = 0xa;
-  genparams.salt[3] = 0xd;
-  genparams.salt[4] = 0xb;
-  genparams.salt[5] = 0xe;
-  genparams.salt[6] = 0xe;
-  genparams.salt[7] = 0xf;
-  genparams.version = 0;
-  genparams.slot_src_sel = 1;
+  // Generate AES output from the boot stage 0 key.
+  gen_params.slot_src_sel = 1;
+  gen_params.sideload_key = true;  // HW key
+  gen_params.key_dest = kDifKeymgrDpeKeyDestAes;
+  gen_params.version = 0;
+  gen_params.salt[7] = 0x945642d9;
+  gen_params.salt[6] = 0xfbcbc925;
+  gen_params.salt[5] = 0xdb7b0691;
+  gen_params.salt[4] = 0xcd973f4d;
+  gen_params.salt[3] = 0x278e051d;
+  gen_params.salt[2] = 0x0d9f1f0d;
+  gen_params.salt[1] = 0x45eff95b;
+  gen_params.salt[0] = 0xb1ad6ba7;
+  generate(&keymgr_dpe, &gen_params);
+  LOG_INFO("KeymgrDpe generated AES output from boot stage 0");
 
-  dif_keymgr_dpe_output_t key;
+  // Derive a boot stage 1 key from the boot stage 0 key.
+  adv_params.slot_src_sel = 1;
+  adv_params.slot_dst_sel = 1;
+  adv_params.max_key_version = 0x100;
+  adv_params.binding_value[7] = 0x4ec9c6d6;
+  adv_params.binding_value[6] = 0x19f5cff7;
+  adv_params.binding_value[5] = 0x426dc745;
+  adv_params.binding_value[4] = 0xb8a8379d;
+  adv_params.binding_value[3] = 0xe92f76e2;
+  adv_params.binding_value[2] = 0xcb68ff71;
+  adv_params.binding_value[1] = 0xaf36e268;
+  adv_params.binding_value[0] = 0xdc96c23d;
+  adv_params.slot_policy = 0x5;
```

Comment:
we should probably put some comment for what policy is being tested (since some policy aspects are not implemented yet)

Reply:
Thx; I added a comment